mkosi/__init__.py

```diff
@@ -2493,8 +2494,35 @@ def invoke_apt(
         INITRD="No",
     )
 
-    run_workspace_command(args, root, cmdline, network=True, env=env)
+    # Ensure apt inside the image can see the local mirror outside of it
+    if args.local_mirror:
+        mirror : Optional[str] = args.local_mirror
+    else:
+        mirror = args.mirror
+    if mirror and mirror.startswith("file:"):
+        nspawn_params = [f"--bind-ro={mirror[5:]}"]
+    else:
+        nspawn_params = None
+
+    run_workspace_command(args, root, cmdline, network=(not args.local_mirror), env=env, nspawn_params=nspawn_params)
```

Comment:
I think the `not args.local_mirror` has the right semantics, but I think this is a bit clearer, since `args.local_mirror=""` doesn't make sense anyway. 
```suggestion
    run_workspace_command(args, root, cmdline, network=(args.local_mirror is None), env=env, nspawn_params=nspawn_params)
```